library/src/otx/data/dataset/anomaly_new.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2023-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Module for OTXSegmentationDataset."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from datumaro.experimental.categories import LabelCategories, LabelSemantic
+
+from otx.data.dataset.base_new import OTXDataset
+from otx.data.entity.sample import AnomalySample
+from otx.types.label import AnomalyLabelInfo
+from otx.types.task import OTXTaskType
+
+if TYPE_CHECKING:
+    from datumaro.experimental import Dataset
+
+
+class OTXAnomalyDataset(OTXDataset):
+    """OTXDataset class for anomaly task."""
+
+    def __init__(self, task_type: OTXTaskType, dm_subset: Dataset, **kwargs) -> None:
```

Comment:
not related to this pr, but looking for your suggestion in general..

here `dm_subset` arg name does not make sense to me as a reader. Is this really a subset? or dataset? Or something else? What are your thoughts?